tests/integration/kafka/test_producer.py

```diff
@@ -71,3 +74,58 @@ def test_partitions_for(self, producer: KafkaProducer, new_topic: NewTopic) -> N
         assert partitions[0].id == 0
         assert partitions[0].replicas == [1]
         assert partitions[0].isrs == [1]
+
+
+@pytest.fixture(scope="function", name="aioproducer")
+async def fixture_aioproducer(
+    kafka_servers: KafkaServers,
+    loop: asyncio.AbstractEventLoop,
+) -> Iterator[AIOKafkaProducer]:
+    try:
+        aioproducer = AIOKafkaProducer(bootstrap_servers=kafka_servers.bootstrap_servers, loop=loop)
+        yield aioproducer
+    finally:
+        await aioproducer.stop()
```

Comment:
```suggestion
    aioproducer = AIOKafkaProducer(bootstrap_servers=kafka_servers.bootstrap_servers, loop=loop)
    try:
        yield aioproducer
    finally:
        await aioproducer.stop()
```

issue: `aioproducer` is not necessarilly successfully assigned to in the `finally` clause unless it happens prior to entering the try-block. This might be input to the other comment about not starting the background thread during instantiation.